src/Shared/PackageManagers/BaseProjectManager.cs

```diff
@@ -364,14 +459,36 @@ public virtual async Task<bool> PackageVersionExistsAsync(PackageURL purl, bool
         }
         
         /// <summary>
-        /// Check if the package version was pulled from the repository.
+        /// Check if the package was pulled from the repository.
         /// </summary>
         /// <param name="purl">The PackageURL to check.</param>
         /// <param name="useCache">If the cache should be checked for the existence of this package.</param>
         /// <returns>True if the package was pulled from the repository. False otherwise.</returns>
+        public virtual Task<bool> PackagePulled(PackageURL purl, bool useCache = true)
+        {
+            return Task.FromResult(false);
+        }
+        
+        /// <summary>
+        /// Check if the package version was pulled from the repository.
+        /// </summary>
+        /// <param name="purl">The PackageURL to check.</param>
+        /// <param name="useCache">If the cache should be checked for the existence of this package.</param>
+        /// <returns>True if the package version was pulled from the repository. False otherwise.</returns>
         public virtual Task<bool> PackageVersionPulled(PackageURL purl, bool useCache = true)
         {
-            throw new NotImplementedException("BaseProjectManager does not implement PackageVersionPulled.");
+            return Task.FromResult(false);
+        }
+        
+        /// <summary>
+        /// Check to see if the package was removed for security.
+        /// </summary>
+        /// <param name="purl">The <see cref="PackageURL"/> to check.</param>
+        /// <param name="useCache">If the cache should be checked for the existence of this package.</param>
+        /// <returns>True if this package is was removed for security. False otherwise.</returns>
+        public virtual Task<bool> PackageRemovedForSecurity(PackageURL purl, bool useCache = true)
+        {
+            return Task.FromResult(false);
         }
```

Comment:
I see what you were going for by having these be helper methods for the above package existence methods, but would anyone actually call these methods? They would probably call the package existence one right, because that one provides any/all such reasons at once?

If callers will probably not use these helper methods, should these just be implementation details on the type-specific managers? I know we would have to move the implementation of the package existence method to the type-specific managers, but that probably makes sense anyway - having these separate method calls do separate JSON parsing does not seem very efficient.

Reply:
Yeah I could have them be implementation specific that makes sense. I don't see a use case for calling this method specifically when the new existence check + record exists.

Reply:
Addressed!